internal/env/env.go

```diff
@@ -139,3 +142,16 @@ func StringSlice(key string, validator func(string) error) ([]string, error) {
 	}
 	return ret, nil
 }
+
+func ParseDuration(dr string) time.Duration {
```

Comment:
This should directly parse a duration from the environment, much like the other functions in this package do it for other data types. 

We need two functions: One that returns a default value, and one that doesn't. So the function prototypes should be:

```go
Duration(key string, validator func(time.Duration) error) (time.Duration, error)
```
and
```go
DurationWithDefault(key string, validator func(time.Duration) error, def time.Duration) time.Duration
```

Reply:
oh, my bad, I misunderstood previous comment and didn't check implementation of other functions for reference.
Addressed all your comments, PTAL